### ISSUE TITLE
When root_dir is not found, cmake-language-server does not work 

### ISSUE THREAD

**Issue by LuoZhongYao** [neovim/nvim-lspconfig#796]
- `nvim --version`: NVIM v0.5.0-dev+1176-g9481310c2
- nvim-lsp version(commit hash): 83d458a0150b3a883454efc0a9ebc4a34162512c
- What language server (If the problem is related to a specific language server): cmake-language-server
- Can you reproduce this behavior on other language server clients (vscode, languageclient-neovim, coc.nvim, etc.): no
- Can you reproduce this behavior on other language servers offered in the nvim-lspconfig repo? (pyls -> pyright): no
- Is the problem isolated to a particular language server: yes
- Operating system/version: Linux localhost.localdomain 5.10.8-100.fc32.x86_64 #1 SMP Sun Jan 17 19:52:43 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux Fedora 32

### How to reproduce the problem from neovim startup
1. Create a directory so that it and its ancestor directory do not have the following files and directories. git,compile_commands.json,build 
2. nvim CmakeLists.txt

### Actual behaviour
No completion 

### Expected behaviour
completion 

### Minimal init.vim or init.lua and code sample
```vim
runtime! debian.vim

call plug#begin(stdpath('config') . '/plugged')
Plug 'Shougo/deoplete.nvim', { 'do': ':UpdateRemotePlugins' }
Plug 'Shougo/deoplete-lsp'
Plug 'lifepillar/vim-solarized8'
Plug 'godlygeek/tabular'
Plug 'neovim/nvim-lspconfig'
call plug#end()

let g:deoplete#enable_at_startup = 1
lua << EOF
local nvim_lsp = require('lspconfig')
local on_attach = function(client, bufnr)
  local function buf_set_keymap(...) vim.api.nvim_buf_set_keymap(bufnr, ...) end
  local function buf_set_option(...) vim.api.nvim_buf_set_option(bufnr, ...) end

  buf_set_option('omnifunc', 'v:lua.vim.lsp.omnifunc')

  -- Mappings.
  local opts = { noremap=true, silent=true }
  buf_set_keymap('n', 'gD', '<Cmd>lua vim.lsp.buf.declaration()<CR>', opts)
  buf_set_keymap('n', 'gd', '<Cmd>lua vim.lsp.buf.definition()<CR>', opts)
  buf_set_keymap('n', 'K', '<Cmd>lua vim.lsp.buf.hover()<CR>', opts)
  buf_set_keymap('n', 'gi', '<cmd>lua vim.lsp.buf.implementation()<CR>', opts)
  buf_set_keymap('n', '<C-k>', '<cmd>lua vim.lsp.buf.signature_help()<CR>', opts)
  buf_set_keymap('n', '<space>wa', '<cmd>lua vim.lsp.buf.add_workspace_folder()<CR>', opts)
  buf_set_keymap('n', '<space>wr', '<cmd>lua vim.lsp.buf.remove_workspace_folder()<CR>', opts)
  buf_set_keymap('n', '<space>wl', '<cmd>lua print(vim.inspect(vim.lsp.buf.list_workspace_folders()))<CR>', opts)
  buf_set_keymap('n', 'gT', '<cmd>lua vim.lsp.buf.type_definition()<CR>', opts)
  buf_set_keymap('n', '<space>r', '<cmd>lua vim.lsp.buf.rename()<CR>', opts)
  buf_set_keymap('n', 'gr', '<cmd>lua vim.lsp.buf.references()<CR>', opts)
  buf_set_keymap('n', '<space>e', '<cmd>lua vim.lsp.diagnostic.show_line_diagnostics()<CR>', opts)
  buf_set_keymap('n', '[d', '<cmd>lua vim.lsp.diagnostic.goto_prev()<CR>', opts)
  buf_set_keymap('n', ']d', '<cmd>lua vim.lsp.diagnostic.goto_next()<CR>', opts)
  buf_set_keymap('n', '<space>q', '<cmd>lua vim.lsp.diagnostic.set_loclist()<CR>', opts)

  -- Set some keybinds conditional on server capabilities
  if client.resolved_capabilities.document_formatting then
    buf_set_keymap("n", "<space>f", "<cmd>lua vim.lsp.buf.formatting()<CR>", opts)
  elseif client.resolved_capabilities.document_range_formatting then
    buf_set_keymap("n", "<space>f", "<cmd>lua vim.lsp.buf.formatting()<CR>", opts)
  end

  -- Set autocommands conditional on server_capabilities
  if client.resolved_capabilities.document_highlight then
    vim.api.nvim_exec ([[
      hi link LspReferenceRead Error
      hi link LspReferenceText Error
      hi link LspReferenceWrite Error
      augroup lsp_document_highlight
        autocmd!
        autocmd CursorHold <buffer> lua vim.lsp.buf.document_highlight()
        autocmd CursorMoved <buffer> lua vim.lsp.buf.clear_references()
      augroup END
    ]], false)
  end
end

-- Use a loop to conveniently both setup defined servers
-- and map buffer local keybindings when the language server attaches
local servers = { "clangd", "cmake", "rls" }
for _, lsp in ipairs(servers) do
  nvim_lsp[lsp].setup { on_attach = on_attach }
end
EOF

```
### Health check
<details>
<summary>Checkhealth result</summary>

health#lspconfig#check
========================================================================
## Checking language server protocol configuration
  - INFO: cmake: configuration checked.
  - ERROR: rls: The given command "rls" is not executable.
  - INFO: clangd: configuration checked.

</details>

### LSP log

<details>
<summary>Log file</summary>

[ START ] 2021-03-26T09:57:59+0800 ] LSP logging initiated
</details>

**Comment by mjlbach** [comment]
I don't understand the issue? If you have nothing to trigger the root directory detection, we have no way to set the root directory (which we have to send to the language server), so we can't start the language server.

**Comment by LuoZhongYao** [comment]
> 
> 
> I don't understand the issue? If you have nothing to trigger the root directory detection, we have no way to set the root directory (which we have to send to the language server), so we can't start the language server.

Root_dir is not detected, the directory where the current file is located should be used as root_dir. And when I use clangd, there is no such problem, clangd can complete the .c  file

**Comment by mjlbach** [comment]
The root directory pattern for cmake-language-server is https://github.com/neovim/nvim-lspconfig/blob/42a13e5ad99b688bc7376fa4d236e924985d04cc/lua/lspconfig/cmake.lua#L8, you must have one of those files in the parent directory of the cmake file you want to find in order to activate the language server.

**Comment by LuoZhongYao** [comment]
> 
> 
> The root directory pattern for cmake-language-server is
> 
> https://github.com/neovim/nvim-lspconfig/blob/42a13e5ad99b688bc7376fa4d236e924985d04cc/lua/lspconfig/cmake.lua#L8
> , you must have one of those files in the parent directory of the cmake file you want to find in order to activate the language server.

When a new project is created and the project is not compiled or added to the version library, LSP support cannot be obtained, which does not conform to the logic of a text editor

**Comment by mjlbach** [comment]
LSP support expects a root directory, and 95% of projects use git. In practice `.git` is a safe assumption. If you disagree, it's a one line PR to add a fallback to the current working director ( and you're welcome to file it).

**Comment by mjlbach** [comment]
closed by #797
